addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -242,6 +255,28 @@ def set_custom_notifications(self, custom_notifications):
         self.custom_notifications = custom_notifications
         self._bus_send_store(self.channel_id, {"custom_notifications": self.custom_notifications})
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_transfer_channel_owner(self):
+        for channel_member in self:
+            # sudo: discuss.channel.member - reading channel ownership related to a member is considered acceptable
+            if channel_member.is_self and len(channel_member.channel_id.channel_partner_ids) > 1 and channel_member.sudo().channel_role == "owner":
```

Comment:
`channel_partner_ids` is deprecated/should not be used. And fetching all members just to count them will not scale. Should use `member_count` instead.